include/zephyr/mgmt/mcumgr/mgmt/mgmt.h

```diff
@@ -174,14 +174,15 @@ typedef void (*mgmt_reset_buf_fn)(void *buf, void *arg);
  *
  * @return 0 if a response was successfully encoded, #mcumgr_err_t code on failure.
  */
-typedef int (*mgmt_handler_fn)(struct smp_streamer *ctxt);
+typedef int (*mgmt_handler_fn)(struct smp_streamer *ctxt, void *user_data);
 
 /**
  * @brief Read handler and write handler for a single command ID.
  */
 struct mgmt_handler {
 	mgmt_handler_fn mh_read;
 	mgmt_handler_fn mh_write;
+	void *user_data;
```

Comment:
I do not really like the addition. The PR does not provide reason for the change and addition. So far we have been able to live without this, while extending support for different commands.

Reply:
Hi @de-nordic, I just added a RFC, please check the comment below